src/Workspaces/Core/Portable/Diagnostics/DiagnosticProvider.cs

```diff
@@ -1,60 +0,0 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-#nullable disable
-
-using System;
-using Microsoft.CodeAnalysis.Diagnostics;
-using Microsoft.CodeAnalysis.SolutionCrawler;
-
-namespace Microsoft.CodeAnalysis
-{
-    /// <summary>
-    /// Provide a way for users to turn on and off analyzing workspace for compiler diagnostics
-    /// </summary>
-    internal static class DiagnosticProvider
-    {
-        public static void Enable(Workspace workspace, Options options)
```

Comment:
Is this what VS Code or VS for Mac are using to enable diagnostics? My worry is we might have to restore this method and take the options but just ignore the options, i.e. delete lines 22-23.

Reply:
i don't see this in Index at all.  @davidwengier do you se any usages of this in vs4mac?

Reply:
went with your approach @jasonmalinowski